pkg/trace/config/apply.go

```diff
@@ -173,9 +173,23 @@ func (c *AgentConfig) applyDatadogConfig() error {
 	if config.Datadog.IsSet("apm_config.log_file") {
 		c.LogFilePath = config.Datadog.GetString("apm_config.log_file")
 	}
+
 	if config.Datadog.IsSet("apm_config.env") {
 		c.DefaultEnv = config.Datadog.GetString("apm_config.env")
+		log.Debugf("setting DefaultEnv to '%s' (from apm_config.env)", c.DefaultEnv)
+	} else if config.Datadog.IsSet("env") {
+		c.DefaultEnv = config.Datadog.GetString("env")
+		log.Debugf("setting DefaultEnv to '%s' (from 'env' config option)", c.DefaultEnv)
+	} else if config.Datadog.IsSet("tags") {
+		for _, tag := range config.Datadog.GetStringSlice("tags") {
+			if strings.HasPrefix(tag, "env:") {
+				c.DefaultEnv = strings.SplitN(tag, "env:", 2)[1]
```

Comment:
This works, but is slightly scary (thinking of panic, even if not possible in the current form). Can we instead do:
```suggestion
				c.DefaultEnv = strings.TrimPrefix(tag, "env:")
```